packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumEventMapper.kt

```diff
@@ -0,0 +1,330 @@
+                        event.error.fingerprint = encodedError["fingerprint"] as? String
+                    }
+
+                    (encodedResult["view"] as? Map<String, Any?>)?.let {
+                        event.view.name = it["name"] as? String
+                        event.view.referrer = it["referrer"] as? String
+                        event.view.url = it["url"] as String
+                    }
+
+                    event
+                }
+            }
+        }
+        mapperPerf.addSample(perf)
+
+        return result
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    internal fun mapLongTaskEvent(event: LongTaskEvent): LongTaskEvent? {
+        var result: LongTaskEvent?
+        val perf = measureNanoTime {
+            var jsonEvent = event.toJson().asFlutterMap()
+            jsonEvent = normalizeExtraUserInfo(jsonEvent)
+
+            result = callEventMapper("mapLongTaskEvent", event, jsonEvent) { encodedResult, event ->
+                if (encodedResult == null) {
+                    null
+                } else {
+                    (encodedResult["view"] as? Map<String, Any?>)?.let {
+                        event.view.name = it["name"] as? String
+                        event.view.referrer = it["referrer"] as? String
+                        event.view.url = it["url"] as String
+                    }
+
+                    event
+                }
+            }
+        }
+        mapperPerf.addSample(perf)
+
+        return result
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun normalizeExtraUserInfo(encodedEvent: Map<String, Any?>): Map<String, Any?> {
+        val reservedKeys = setOf("email", "id", "name")
+        // Pull out user information
+        val mutableEvent = encodedEvent.toMutableMap()
+        (mutableEvent["usr"] as? Map<String, Any?>)?.let { usr ->
+            val mutableUsr = usr.toMutableMap()
+            val extraUserInfo = mutableMapOf<String, Any?>()
+            usr.filter { !reservedKeys.contains(it.key) }.forEach {
+                extraUserInfo[it.key] = it.value
+                mutableUsr.remove(it.key)
+            }
+            mutableUsr["usr_info"] = extraUserInfo
+            mutableEvent["usr"] = mutableUsr
+        }
+
+        return mutableEvent
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    internal fun <T> callEventMapper(
+        mapperName: String,
+        event: T,
+        encodedEvent: Map<String, Any?>,
+        completion: (Map<String, Any?>?, T) -> T?
+    ): T? {
+        var modifiedJson: Map<String, Any?>? = encodedEvent
+        // Any valid channel should do here since they should be connected to the same initialization code,
+        // but calling on multiple could result in weird behavior and performance issues. While "first" may
+        // not be the actual first registered channel, it should be fine for our purposes.
+        val channel = channels.first()
```

Comment:
should we handle a situation when there is no channels registered by some reason?